controllers/helmreleaseproxy/helmreleaseproxy_controller.go

```diff
@@ -234,7 +237,29 @@ func (r *HelmReleaseProxyReconciler) reconcileNormal(ctx context.Context, helmRe
 		})
 	}
 
-	release, err := client.InstallOrUpgradeHelmRelease(ctx, kubeconfig, helmReleaseProxy.Spec)
+	credentialsPath := ""
+	if helmReleaseProxy.Spec.CredentialsPath != "" {
+		log.V(2).Info("Setting OCI credentials")
+		credentialsPath = helmReleaseProxy.Spec.CredentialsPath
+	}
+
+	if helmReleaseProxy.Spec.CredentialsSecretRef != nil {
+		log.V(2).Info("Setting OCI credentials")
+		credentialsValues, err := getCredentialsFromSecret(ctx, helmReleaseProxy.Spec.CredentialsSecretRef.Name, helmReleaseProxy.Spec.CredentialsSecretRef.Namespace, r.Client)
```

Comment:
Instead of passing `r.Client` as a parameter why not change `getCredentialsFromSecret` to a receiver function such as `(r *HelmReleaseProxyReconciler)getCredentialsFromSecret()` and directly access the client, for example via `r.Client.Get()`